types/networks/src/lib.rs

```diff
@@ -302,3 +301,108 @@ impl Default for ChainConfig {
         }
     }
 }
+
+pub fn default_network_version() -> NetworkVersion {
+    NetworkVersion::V1
+}
+
+pub mod de_network_version {
+    use fil_types::NetworkVersion;
+    use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<NetworkVersion, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let version: &str = Deserialize::deserialize(deserializer)?;
+        let version = version.to_lowercase();
+
+        match version.as_str() {
+            "v0" => Ok(NetworkVersion::V0),
+            "v1" => Ok(NetworkVersion::V1),
+            "v2" => Ok(NetworkVersion::V2),
+            "v3" => Ok(NetworkVersion::V3),
+            "v4" => Ok(NetworkVersion::V4),
+            "v5" => Ok(NetworkVersion::V5),
+            "v6" => Ok(NetworkVersion::V6),
+            "v7" => Ok(NetworkVersion::V7),
+            "v8" => Ok(NetworkVersion::V8),
+            "v9" => Ok(NetworkVersion::V9),
+            "v10" => Ok(NetworkVersion::V10),
+            "v11" => Ok(NetworkVersion::V11),
+            "v12" => Ok(NetworkVersion::V12),
+            "v13" => Ok(NetworkVersion::V13),
+            "v14" => Ok(NetworkVersion::V14),
+            "v15" => Ok(NetworkVersion::V15),
+            _ => Err(de::Error::custom(&format!(
+                "Invalid network version: {}",
+                version
+            ))),
+        }
+    }
+
+    pub fn serialize<S>(n: &NetworkVersion, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        n.to_owned().serialize(serializer)
+    }
+}
+
+#[cfg(test)]
+pub mod test {
+    use super::*;
+
+    fn remove_whitespace(s: &mut String) {
```

Comment:
It'd be probably cleaner to have a method that returns a copy without whitespace instead of mutating the parameter. Then we could get rid of all the `mut`s.